pkg/node/manager/manager.go

```diff
@@ -90,14 +93,47 @@ type Manager struct {
 	metricDatapathValidations prometheus.Counter
 }
 
+// Subscribe subscribes the given node handler to node events.
+func (m *Manager) Subscribe(nh datapath.NodeHandler) {
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+	m.nodeHandlersMu.Lock()
+	m.nodeHandlers[nh] = struct{}{}
+	m.nodeHandlersMu.Unlock()
+	// Add all nodes already received by the manager.
+	for _, v := range m.nodes {
+		v.mutex.Lock()
+		nh.NodeAdd(v.node)
+		v.mutex.Unlock()
+	}
+}
+
+// Unsubscribe unsubscribes the given node handler with node events.
+func (m *Manager) Unsubscribe(nh datapath.NodeHandler) {
+	m.nodeHandlersMu.Lock()
+	delete(m.nodeHandlers, nh)
+	m.nodeHandlersMu.Unlock()
+}
+
+// Iter executes the given function in all subscribed node handlers.
```

Comment:
Does `m.mutex` need to be held to call `m.Iter(...)`? (`Subscribe()` has this locking pattern)

If so, worth adding to the comment. That said, `m.mutex` will only be accessible within the package, so then should `Iter()` be exported? Or does there need to be two, one for users external to this package and one for internal users which already hold `m.mutex`?

Reply:
See [1]